### PR TITLE
[CHK-415] Refactor PSP filters

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes

- Update PSP repository with `@Query` annotation
- Fix PSP filtering
- Add unit tests for PSP filtering

#### How Has This Been Tested?
- Unit, local run

- docker-compose build/run
#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -122,7 +122,7 @@
 							<goal>generate</goal>
 						</goals>
 						<configuration>
-							<inputSpec>https://raw.githubusercontent.com/pagopa/pagopa-ecommerce-payment-instruments-service/test-helm-deploy/api-config-mock/openapi.json</inputSpec>
+							<inputSpec>https://raw.githubusercontent.com/pagopa/pagopa-ecommerce-payment-instruments-service/main/api-config-mock/openapi.json</inputSpec>
```
Comment (infantesimone) [inline pom.xml:125]: you can use swagger of api-config repo. after this PR https://github.com/pagopa/pagopa-api-config/pull/125

File: src/main/java/it/pagopa/ecommerce/payment/instruments/infrastructure/rule/FilterRuleEngine.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import it.pagopa.ecommerce.payment.instruments.infrastructure.PspDocument;
+import it.pagopa.ecommerce.payment.instruments.infrastructure.PspRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+@AllArgsConstructor
+public class FilterRuleEngine {
```
Comment (infantesimone) [inline src/main/java/it/pagopa/ecommerce/payment/instruments/infrastructure/rule/FilterRuleEngine.java:12]: what do you think if `FilterRuleEngine` became a spring component and use it with `@Autowired`?

File: pom.xml (hunk 1 above)
Comment (MarcoAGreco) [reply pom.xml]: Great, solved with [b87b662](https://github.com/pagopa/pagopa-ecommerce-payment-instruments-service/pull/9/commits/b87b66264638b1bcdcb27d427c2992de9b8e36d8) 👍

**Comment by infantesimone** [comment]
@MarcoAGreco great refactoring! can you add some unit tests related to the new code?

**Review by infantesimone — APPROVED** [review]

**Pull request merged**